### ISSUE TITLE
[vertical metrics] issues with alignment across browsers / platforms

### ISSUE THREAD

**Issue by dalemartyn** [IBM/plex#254]
I've noticed that the alignment is inconsistent when using different browsers. The main Plex Sans complete TTF seems to work fine everywhere, but the woff2s (complete and split) and variable fonts have issues in Safari, Firefox and Chrome OS.

The Google Fonts version seems to be fine, as does iA writer's variable fonts that are based on plex.

I've uploaded a demo to https://dalemartyn.github.io/ibmplex_vmet/ along with the metrics that I've looked up using FontForge.

### Chrome on macOS

<img width="566" alt="Chrome on macOS" src="https://user-images.githubusercontent.com/10400154/57980259-ce49dc80-7a20-11e9-91b0-78277c647b36.png">

### Safari on macOS

<img width="563" alt="Safari on macOS" src="https://user-images.githubusercontent.com/10400154/57980242-9773c680-7a20-11e9-9545-34b0be7aa259.png">

### Firefox on macOS

<img width="558" alt="Firefox on macOS" src="https://user-images.githubusercontent.com/10400154/57980246-a0fd2e80-7a20-11e9-9fcb-f7625bde438a.png">

### Chrome on Chrome OS

<img width="574" alt="Chrome OS" src="https://user-images.githubusercontent.com/10400154/57980250-a9ee0000-7a20-11e9-97bd-86a15634c753.png">

**Comment by dalemartyn** [comment]
I think that the issue is that the baseline is too high when the typo metrics are being used. (and apparently variable fonts [require the use of typo metrics](https://github.com/googlefonts/fontmake/issues/492)).

I've tried setting the metrics to Typo Ascent 880 and Typo Descent -120 (complete guesses) and it seems much better.

<img width="461" alt="Firefox on macOS" src="https://user-images.githubusercontent.com/10400154/57982194-eed26080-7a39-11e9-8c37-d856da6145db.png">

Link to demo: https://dalemartyn.github.io/ibmplex_vmet_fix/

**Comment by BoldMonday** [comment]
Thank you for the thorough comparison. The differences in vertical metrics in various environments are not intended and we will investigate this further.

Changing certain settings to smaller values as you tried yourself might look okay if you only focus on basic uppercase and lowercase letters. However, the fonts also include glyphs with larger vertical dimensions such as letters with (stacked) diacritics. Those glyphs will be clipped then.

**Comment by dalemartyn** [comment]
Yes, I agree. I'm not an expert so hoping someone with more knowledge can investigate further! Thank you :-)

**Comment by mjabbink** [comment]
@BoldMonday Any insights yet in this issue?

**Comment by garcia323** [comment]
@BoldMonday Is this issue being addressed? I'm currently using the Google Fonts version for development since I'm having issues with vertical alignment in IE11 when I use the fonts in Web.zip.

**Comment by mjabbink** [comment]
This happens with Hebrew as well just between weights. The medium has different vertical measurement in Adobe illustrator for example.

**Comment by mjabbink** [comment]
@BoldMonday have you had a chance to investigate this?

**Comment by vpicone** [comment]
@BoldMonday @mjabbink is there any way for us to prioritize this effort? I've encountered this issue as well and are starting to see others speak up too.

**Comment by mjabbink** [comment]
@BoldMonday this is a serious issue. I also think it’s happening in Hebrew across weights. Medium weight has different vertical height than the Regular for example. We should prioritize this update. We can plan in next meeting.

**Comment by mjabbink** [comment]
@BoldMonday was there any insights on this? Do we need to investigate further?

**Comment by BoldMonday** [comment]
@mjabbink Last time we agreed about creating an environment with the web dev team where we can test these things directly. We want to be able to see the same things as you guys.

**Comment by garcia323** [comment]
I found a workaround. IBM Plex is available through Google Fonts and these don't have the vertical alignment issues. Since we can't ship our product with links to Google Fonts, I downloaded the font definitions (WOFF and WOFF2) from Google using this site:  http://google-webfonts-helper.herokuapp.com/fonts/ibm-plex-sans?subsets=latin

**Comment by francescomariani** [comment]
Hi @BoldMonday and @mjabbink ,
We are in the process of adopting IBM Plex and see this issue across the whole application and browser, representing a showstopper for us.

Since this issue has been opened since May 2019 (and it is already fixed in the Google repository mentioned above), is there any plan and timeline to get a fix?

The issue is also visible in our IBM Plex public website
https://www.ibm.com/plex/specs/

Thanks

**Comment by grahamharper** [comment]
Some screenshots of the rendering of the Plex site in IE11 and Chrome on Windows.

![ibm plex website - ie11](https://user-images.githubusercontent.com/2853295/72805223-428ced80-3c4a-11ea-9898-cb873faf29f7.png)
![ibm plex website - chrome](https://user-images.githubusercontent.com/2853295/72805227-46207480-3c4a-11ea-8c92-ee70faa86e43.png)

Why does the Google Fonts hosted version of Plex not exhibit this issue but the version downloaded from GitHub does?

**Comment by grahamharper** [comment]
Some screenshots of the rendering of the Plex site in IE11 and Chrome on Windows.

![ibm plex website - ie11](https://user-images.githubusercontent.com/2853295/72805460-d363c900-3c4a-11ea-8c40-61e25b3d383c.png)
![ibm plex website - chrome](https://user-images.githubusercontent.com/2853295/72805462-d5c62300-3c4a-11ea-87b1-c21cb4928467.png)


Why does the Google Fonts hosted version of Plex not exhibit this issue but the version downloaded from GitHub does?

**Comment by mjabbink** [comment]
@BoldMonday

**Comment by BoldMonday** [comment]
@mjabbink In order to solve this properly we need a testing environment. (See my previous comment.)
The issue here is larger than only web browsers because the fonts need to work properly in applications on desktop systems too. Google does not need to take care of that.

**Comment by mjabbink** [comment]
@vpicone or @joshblack can potentially help here too. Having test environments set up for Plex testing would be hugely helpful for much of the issues we see

**Comment by BoldMonday** [comment]
New settings for vertical metrics have been applied in the latest (v.5.0.0) release. This has been tested and approved by a team of Watson developers.

I believe the issue can be closed now.

**Comment by dalemartyn** [comment]
Hi @BoldMonday 
thank you for fixing this 👍 
are there any plans to add the new metrics to the Variable fonts?

**Comment by ddegan** [comment]
Hi! As @dalemartyn pointed out, are there any plans to add the same fix to the variable font?
Thank you

**Comment by kgibb** [comment]
Is it possible to fix the variable font also please?

**Comment by mjabbink** [comment]
@vpicone is there a testing environment we can create for @BoldMonday. I don't think we created this or did we?

**Comment by leerob** [comment]
Related: https://github.com/IBM/plex/issues/377

FYI, I solved this with a workaround using https://seek-oss.github.io/capsize/
